resolver/resolver.cc

```diff
@@ -1298,7 +1298,7 @@ class ResolveConstantsWalk {
         nesting_ = nesting_->parent;
     }
 
-    const bool checkAmbiguousDefinition(core::Context ctx) {
+    const bool checkAmbiguousDefinition(core::Context ctx, core::SymbolRef curSym, shared_ptr<Nesting> curNesting) {
```

Comment:
Can we take the `shared_ptr` by reference here so we're not constantly doing atomic reference counts unnecessarily?

```suggestion
    const bool checkAmbiguousDefinition(core::Context ctx, core::SymbolRef curSym, const shared_ptr<Nesting> &curNesting) {
```